src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceSetDeserializer.cs

```diff
@@ -48,7 +49,7 @@ public override object Read(ODataMessageReader messageReader, Type type, ODataDe
                 throw Error.Argument("edmType", SRResources.ArgumentMustBeOfType, EdmTypeKind.Complex + " or " + EdmTypeKind.Entity);
             }
 
-            ODataReader resourceSetReader = messageReader.CreateODataResourceSetReader();
+            ODataReader resourceSetReader = readContext.IsChangedObjectCollection ? messageReader.CreateODataDeltaResourceSetReader() : messageReader.CreateODataResourceSetReader();
```

Comment:
>readContext.IsChangedObjectCollection  [](start = 44, length = 38)

We should have a ODataDeltaResourceSetDeserializer  to process the Delta ResourceSet deserializer